### PR TITLE
Added GL_CLAMP_TO_BORDER support

### PR DESCRIPTION
**Pull request by Lectem**
Everything is in the title !

### CODE REVIEW

**Comment by neobrain** [comment]
How was this tested?

**Comment by Lectem** [comment]
It was tested with https://github.com/Lectem/3DS_gpu_tests/blob/wrap_modes .
There was an issue when using the u/v coordinates that were >= to the height/width, I didn't see it because I only mapped my texture to a square with coordinates -0.5 -0.5 0.5 0.5 at first. I fixed the code (the upper bounds check was wrong).

**Comment by neobrain** [comment]
@Lectem: That link doesn't work.

**Comment by Lectem** [comment]
Sorry, I updated it

File: src/video_core/rasterizer.cpp [hunk 1]
```diff
@@ -367,17 +370,26 @@ static void ProcessTriangleInternal(const VertexShader::OutputVertex& v0,
...
-                texture_color[i] = DebugUtils::LookupTexture(texture_data, s, t, info);
-                DebugUtils::DumpTexture(texture.config, texture_data);
+                if ( (texture.config.wrap_s == Regs::TextureConfig::ClampToBorder && (s < 0 || s >= texture.config.width))
+                    || (texture.config.wrap_t == Regs::TextureConfig::ClampToBorder && (t < 0 || t >= texture.config.height)) )
+                {
+                    auto border_color = texture.config.border_color;
+                    texture_color[i] = { border_color.r, border_color.g, border_color.b, border_color.a };
+                }
+                else
+                {
```
Comment (bunnei) [inline src/video_core/rasterizer.cpp]: Please see our style guide (https://github.com/citra-emu/citra/blob/master/CONTRIBUTING.md) for guidance on where the `{` belongs.

Comment (Lectem) [inline src/video_core/rasterizer.cpp]: Fixed

File: src/video_core/rasterizer.cpp [hunk 2]
```diff
@@ -367,17 +370,23 @@ static void ProcessTriangleInternal(const VertexShader::OutputVertex& v0,
...
-                // TODO: Check if this applies to ETC textures, too.
-                s = GetWrappedTexCoord(texture.config.wrap_s, s, texture.config.width);
-                t = texture.config.height - 1 - GetWrappedTexCoord(texture.config.wrap_t, t, texture.config.height);
-
-                u8* texture_data = Memory::GetPhysicalPointer(texture.config.GetPhysicalAddress());
-                auto info = DebugUtils::TextureInfo::FromPicaRegister(texture.config, texture.format);
-
-                texture_color[i] = DebugUtils::LookupTexture(texture_data, s, t, info);
-                DebugUtils::DumpTexture(texture.config, texture_data);
+                if ( (texture.config.wrap_s == Regs::TextureConfig::ClampToBorder && (s < 0 || s >= texture.config.width))
```
Comment (lioncash) [inline src/video_core/rasterizer.cpp]: space after the first `(` shouldn't be there.

Comment (Lectem) [inline src/video_core/rasterizer.cpp]: Fixed, I find it harder to read though.

**Comment by bunnei** [comment]
LGTM

**Comment by neobrain** [comment]
Please make sure to keep the test repository alive until we migrate the test to a more central place.

**Pull request merged**